test/rfc7540_SUITE.erl

```diff
@@ -375,6 +375,36 @@ lingering_data_counts_toward_connection_window(_) ->
 	timer:sleep(300),
 	gun:close(ConnPid).
 
+respect_max_concurrent_streams(_) ->
+	doc("The SETTINGS_MAX_CONCURRENT_STREAMS setting can be used to "
+		"restrict the number of concurrent streams. (RFC7540 5.1.2, RFC7540 6.5.2)"),
+	{ok, Ref, Port} = do_cowboy_max_councurrent_streams(1),
+	try
+		{ok, ConnPid} = gun:open("localhost", Port, #{protocols => [http2]}),
+		{ok, http2} = gun:await_up(ConnPid),
+		StreamRef1 = gun:get(ConnPid, "/"),
+		timer:sleep(100),
+		StreamRef2 = gun:get(ConnPid, "/"),
+		{error, {stream_error, Reason}} = gun:await(ConnPid, StreamRef2),
+		too_many_streams = Reason,
+		{response, nofin, 200, _} = gun:await(ConnPid, StreamRef1),
+		{ok, _} = gun:await_body(ConnPid, StreamRef1),
+		gun:close(ConnPid)
+	after
+		cowboy:stop_listener(Ref)
+	end.
+
+do_cowboy_max_councurrent_streams(MaxConcurrentStreams) ->
+	Ref = make_ref(),
+	Routes = [{'_', [{"/", delayed_hello_h, 500}]}],
```

Comment:
Let's change `"/"` to `"/delayed"` to make it more obvious.

Also we probably don't need to do the Cowboy init in a separate function, we can do it in the test directly. The other function `do_cowboy_origin` is used in multiple places that's why it's separate.

Reply:
Done.